ArduPlane/commands_logic.cpp

```diff
@@ -16,22 +16,22 @@ bool Plane::start_command(const AP_Mission::Mission_Command& cmd)
     // special handling for nav vs non-nav commands
     if (AP_Mission::is_nav_cmd(cmd)) {
         // set land_complete to false to stop us zeroing the throttle
-        auto_state.land_complete = false;
-        auto_state.land_pre_flare = false;
+        landing.complete = false;
```

Comment:
This should be wrapped in a helper function inside AP_Landing, rather then allowing plane to manage AP_Landing's state, same for the other states manipulated below.